src/Task.ts

```diff
@@ -438,159 +439,78 @@ export class Task {
-
-        // If the task is a block quote, the block quote wraps the p-tag that contains the content.
-        // In that case, we need to unwrap the p-tag *inside* the surrounding block quote.
-        // Otherwise, we unwrap the p-tag as a direct descendant of the textSpan.
-        const blockQuote = textSpan.querySelector('blockquote');
-        const directParentOfPTag = blockQuote ?? textSpan;
-
-        // Unwrap the p-tag that was created by the MarkdownRenderer:
-        const pElement = directParentOfPTag.querySelector('p');
-        if (pElement !== null) {
-            while (pElement.firstChild) {
-                directParentOfPTag.insertBefore(pElement.firstChild, pElement);
-            }
-            pElement.remove();
-        }
-
-        // Remove an empty trailing p-tag that the MarkdownRenderer appends when there is a block link:
-        textSpan.findAll('p').forEach((pElement) => {
-            if (!pElement.hasChildNodes()) {
-                pElement.remove();
-            }
-        });
-
-        // Remove the footnote that the MarkdownRenderer appends when there is a footnote in the task:
-        textSpan.findAll('.footnotes').forEach((footnoteElement) => {
-            footnoteElement.remove();
-        });
-
-        const checkbox = li.createEl('input');
-        checkbox.addClass('task-list-item-checkbox');
-        checkbox.type = 'checkbox';
-        if (this.status !== Status.TODO) {
-            checkbox.checked = true;
-            li.addClass('is-checked');
-        }
-        checkbox.onClickEvent((event: MouseEvent) => {
-            event.preventDefault();
-            // It is required to stop propagation so that obsidian won't write the file with the
-            // checkbox (un)checked. Obsidian would write after us and overwrite our change.
-            event.stopPropagation();
-
-            // Should be re-rendered as enabled after update in file.
-            checkbox.disabled = true;
-            const toggledTasks = this.toggle();
-            replaceTaskWithTasks({
-                originalTask: this,
-                newTasks: toggledTasks,
-            });
-        });
-
-        li.prepend(checkbox);
-
-        // Set these to be compatible with stock obsidian lists:
-        li.setAttr('data-task', this.originalStatusCharacter.trim()); // Trim to ensure empty attribute for space. Same way as obsidian.
-        li.setAttr('data-line', listIndex);
-        checkbox.setAttr('data-line', listIndex);
-
-        if (layoutOptions?.shortMode) {
-            this.addTooltip({ element: textSpan, isFilenameUnique });
-        }
-
-        return li;
+    /**
+     * Create an HTML rendered List Item element (LI) for the current task.
+     * @param {renderTails}
+     */
+    public async toLi(renderDetails: TaskLineRenderDetails): Promise<HTMLLIElement> {
+        return renderTaskLine(this, renderDetails);
```

Comment:
There are some references to `Task.toLi` in CONTRIBUTING.

Could you please have a review of them and see if anything needs updating with the removal of code out to `renderTaskLine()`?

Reply:
Will do.